using_the_compiler/README.md

```diff
@@ -20,7 +20,7 @@ Alternatively you can use the `run` command:
 $ crystal run some_program.cr
 ```
 
-## Creating an executable
+## Creating a dynamically-linked executable
```

Comment:
This change is unrelated to binary size reduction, please send it in a separate PR.